### ISSUE TITLE
zprint deletes lines

### ISSUE THREAD

**Issue by atdixon** [kkinnear/zprint#187]
Testing w/ `zprint-1.1.1`.

```
{:list {:respect-nl? true}}
```

```
(def
  ^{:meta :x}
  ; one
  ; two
  ; three
  :body)
```

If you run zprint on this code, it will remove the "three" comment. If you run it again, it removes the "two" comment.

**Comment by kkinnear** [comment]
Thank you so much for reporting this!  Indeed, comments (or any newlines) in between meta-data and the object of the meta-data are mishandled, resulting in missing some of them. 

While any bug that loses data is very serious, at least in this case the only loss is between the meta-data and the thing the meta-data is attached to.  The meta-data and the element to which it is attached are not involved in this particular bug.  

The origin of this bug is that there is a separate indent specified for each of the "things" between the meta data and the element to which it is attached, and the routine that creates the indents only creates them for the non-newline elements.  But the routine that interprets them expects there to be one for every element, including newlines, even if they aren't used for the newlines.  Every comment has a separate newline following it.  So the count of indents falls short of the elements to be formatted, and the formatting stops because the formatting is happening in a map over several collections, and the map terminates when the shortest collection is consumed.  This doesn't affect the thing to which the meta-data is attached because it isn't involved in this map operation, it is formatted separately because it is the "rightmost" element in the collection.

Unfortunately, there isn't a workaround for this bug.  It will be fixed in version `1.1.2`.  Until then, don't put comments between
the meta-data and the object to which it is attached.

Thanks again for noticing this, and taking the time to report it.  I really appreciate it!

**Comment by kkinnear** [comment]
This is fixed in 1.1.2:
```
zprint.core=> (print i187)
(def
  ^{:meta :x}
  ; one
  ; two
  ; three
  :body)
nil
zprint.core=> (czprint i187 {:parse-string? true})
(def ^{:meta :x}
     ; one
     ; two
     ; three
     :body)
```
Thanks again for reporting this!
